From 3da466d9a15d91476ebda2ccd89412ba9efe3bb1 Mon Sep 17 00:00:00 2001
From: Lukas Slebodnik <slebodnik@sors.com>
Date: Thu, 22 Mar 2012 16:31:33 +0100
Subject: [PATCH 1/3] Fixed parsing condor_status command line arguments.
 -attributes and -target requires only 1 argument, but 2 arguments was skipped.

---
 doc/man-pages/condor_status.tex |    2 +-
 src/condor_status.V6/status.cpp |   12 +++++++++---
 2 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/doc/man-pages/condor_status.tex b/doc/man-pages/condor_status.tex
index 1888380..18a0e25 100644
--- a/doc/man-pages/condor_status.tex
+++ b/doc/man-pages/condor_status.tex
@@ -98,7 +98,7 @@ options} may be specified.
       the state of machine, the virtual machine memory,
       and the type of networking.}
 
-    \OptItem{\OptArg{-attributes}{Attr1 [,Attr2 \Dots]}}{(Display option)
+    \OptItem{\OptArg{-attributes}{Attr1[,Attr2 \Dots]}}{(Display option)
       Explicitly list the attributes in a comma separated list
       which should be displayed when using the \Opt{-xml} or \Opt{-long}
       options.  Limiting the number of attributes increases the efficiency 
diff --git a/src/condor_status.V6/status.cpp b/src/condor_status.V6/status.cpp
index 4ebf7c9..74fa0fc 100644
--- a/src/condor_status.V6/status.cpp
+++ b/src/condor_status.V6/status.cpp
@@ -627,7 +627,13 @@ firstPass (int argc, char *argv[])
 			setPPstyle (PP_XML, i, argv[i]);
 		} else
 		if (matchPrefix (argv[i],"-attributes", 3)){
-			// we don't do anything right here ... see prefix check in secondPass
+			if( !argv[i+1] ) {
+				fprintf( stderr, "%s: -attributes requires one additional argument\n",
+						 myName );
+				fprintf( stderr, "Use \"%s -help\" for details\n", myName );
+				exit( 1 );
+			}
+			i++;
 		} else	
 		if (matchPrefix (argv[i], "-run", 2) || matchPrefix(argv[i], "-claimed", 3)) {
 			setMode (MODE_STARTD_RUN, i, argv[i]);
@@ -814,7 +820,7 @@ secondPass (int argc, char *argv[])
 			continue;
 		}
 		if (matchPrefix (argv[i], "-target", 2)) {
-			i += 2;
+			i++;
 			continue;
 		}
 		if( matchPrefix(argv[i], "-sort", 3) ) {
@@ -832,7 +838,7 @@ secondPass (int argc, char *argv[])
 			while( (s=more_attrs.next()) ) {
 				projList.AppendArg(s);
 			}
-			i += 2;
+			i++;
 			continue;
 		}
 		
-- 
1.7.4.1

